### ISSUE TITLE
Support beta features

### ISSUE THREAD

**Issue by rgaudin** [offspot/imager-service#399]
There are now many projects composing the Hotspot at large with a _single_ general UI that's used both by us and end-users to create images.

What would you think of allowing Cardshop users to _test_ new features?

For instance, I have a new dashboard version that I want to be tested thoroughly before it being rolled out in everybody's cards. The most convenient way would be to have a place in the Cardshop configurator to select _in-testing features_ where one could select “dashboard 1.4”.

Additional question: should this be visible to Admins only or all users? All users is an advantage for testing but we're following an aggressively minimalist strategy…
An option there could be to have “Enable experimental features” option or something that would toggle it. We have no preference pane for users so it would an entry in user menu (next to Password reset) for now.

**Comment by rgaudin** [comment]
@Popolechien @kelson42 @benoit74 ; opinion requested

**Comment by Popolechien** [comment]
> we're following an aggressively minimalist strategy

It would be even more aggressive if I had my way!

My take here is that yes, we should allow this, but no, we shouldn't make it available to all users by default. There needs to be a mechanism whereby participants would _actively and knowingly_ decide to participate / use. If only because since we can't track behaviour we'd have to make sure that they can open tickets and report issues in a useful manner

**Comment by rgaudin** [comment]
> There needs to be a mechanism

Would the one described work for you? In the user menu, a link to “Experimental features” or something that leads to a page explaining what it entails ; how to provide feedback (that's very important thank you) and a way to toggle the feature.

**Comment by Popolechien** [comment]
Yeah I think `Beta` should be the term, then there should be a "I have read and understood" tick page, then on to the beta features with a clear reminder somewhere of where to post comments. 

There should also be a way to easily roll that back, for instance by unticking the Beta option. We could slide the whole thing in the profile tab (between name, change password and sign out).

**Comment by rgaudin** [comment]
I get the feeling that nobody wants this feature. My bad for proposing a solution instead of just describing the problem. Seems obvious to me though. We (obviously) want to test new features before deploying them.

If we agree that we can't go from developer to users directly (if we don't, speak up!), then I imagine the following options:

- testers need to run `image-creator` locally, adapting `image.yaml` and all. **Not feasible**
- we setup a separate imager-service just for tests: backend, db, frontend, frontend-db, worker, etc. Waste of resources, additional infra burden, no integration with existing configs and all and… since it'd be a regular imager, only possible to test one particular thing at once ; and with an additional burden for dev to maintain this separate thing. **hardly maintainable, less flexible**
- setup another online demo with the features to test. Apart from it not being ready, it would require duplicating the demo, without the possibility to test everything that's not in the demo: WIFi, captive portal, image, etc. It would be limited to a single test at a time, etc. **hardly maintainable, less flexible**
- What I proposed above.

I don't see much down sides but I am certainly biased

**Comment by benoit74** [comment]
Oh, sorry, I have badly expressed my feeling, let me rephrase this: **every Kiwix/openZIM user absolutely need this feature**. I don't have hard feelings / opinions about the additional question regarding other users

**Comment by Popolechien** [comment]
> I get the feeling that nobody wants this feature.

I disagree with that statement. I rather think that the options/alternatives/business case weren't clearly laid out. Both @benoit74 and myself agreed (or did not disagree), with the only express condition that it is not to be offered or rolled out to all users (as in, they must be actively seeking to activate it). 

Other than I am unclear on the amount of work required, the timing, and as to what the current roadmap for future improvements is that would warrant developing this feature already (as opposite to doing it later when there's an imager 3.0 in the works, or when our user base isn't as puny anymore as to not allow anymore for the current push>test>comment>fix process). That's more my general feeling.

**Comment by rgaudin** [comment]
OK thank you both for clarifying this. I think it's lightweight to implement but haven't looked at it in detail. Wanted to make sure we're all on the same page since it's a bit against our own way of doing (using prod resources for devel) before looking at it in detail.
I think it may happen really soon because every new change/fix/feature is difficult to test even for me without this.
